poetry.lock

```diff
@@ -437,7 +437,7 @@ name = "py4j"
 version = "0.10.9"
 description = "Enables Python programs to dynamically access arbitrary Java objects"
 category = "main"
-optional = true
+optional = false
```

Comment:
Any context on this change?

Reply:
Nice catch! I think I changed `pyspark` to become a required dependency for local development purposes, which probably made `py4j` a required dependency as well. Probably just need to re-generate the **poetry.lock** file to reflect what is defined on pyproject.toml.

Thanks for the heads up!